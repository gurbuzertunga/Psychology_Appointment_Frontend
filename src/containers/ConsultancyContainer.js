import React, { useState } from 'react';
import { connect } from 'react-redux';
import { useParams, Redirect, useHistory } from 'react-router-dom';
import AOS from 'aos';
import 'aos/dist/aos.css';
import PropTypes from 'prop-types';
import { setAppointment } from '../actions';
import makeAppointment from '../requests/makeAppointment';

AOS.init({
  delay: 200, // values from 0 to 3000, with step 50ms
  duration: 400, // values from 0 to 3000, with step 50ms
});
const ConsultancyContainer = ({ consultancies, authToken, setAppointment }) => {
  const { id } = useParams();
  if (!authToken) {
    return <Redirect to="/" />;
  }
  const data = {
    time: '',
    problem: '',
    date: '',
    consultancy_id: parseInt(id), // eslint-disable-line
  };

  const [state, setState] = useState(data);
  const history = useHistory();
  const handleInputChange = e => {
    const appointmentReq = { ...state };
    appointmentReq[e.target.name] = e.target.value;
    setState(appointmentReq);
  };

  const handleSubmit = e => {
    e.preventDefault();
    makeAppointment({ ...state, authToken }, setAppointment);
    setTimeout(() => {
      history.push('/');
    }, 1000);
  };
  return (
    <div className="flex mt-16 justify-evenly">
      <div data-aos="fade-down" className="bg-transparent border border-gray-300 p-4 rounded w-5/12">
        <h1 className="text-white text-2xl font-bold my-4">Consultancy Details</h1>
        {
          consultancies.filter(consultancy => consultancy.id === parseInt(id)).map(el => ( // eslint-disable-line
            <div key={el.id}>
              <h4 className="text-white border-b-2 border-white w-max mt-4">{el.area}</h4>
              <p className="text-white font-light text-sm mt-4">{el.details}</p>
            </div>
          ))
        }
      </div>
      <form className="flex flex-col border border-gray-300 rounded py-4" onSubmit={handleSubmit}>
        <input className="font-light focus:outline-none focus:ring focus:border-blue-300 bg-transparent bolder placeholder-white text-white p-1 mb-4 mx-6 border border-gray-300 rounded" type="time" name="time" id="time" onChange={handleInputChange} />
        <input className="font-light focus:outline-none focus:ring focus:border-blue-300 bg-transparent bolder placeholder-white text-white p-1 mb-4 mx-6 border border-gray-300 rounded" type="date" name="date" id="date" onChange={handleInputChange} />
        <textarea className="font-light text-sm focus:outline-none focus:ring focus:border-blue-300 bg-transparent bolder placeholder-white text-white p-1 mb-4 mx-6 border border-gray-300 rounded" name="problem" id="problem" placeholder="Some details about your problem..." onChange={handleInputChange} />
        <button className="text-white font-light" type="submit">Submit</button>
      </form>
    </div>
  );
};
ConsultancyContainer.propTypes = {
  consultancies: PropTypes.arrayOf(Object).isRequired,
  authToken: PropTypes.string,
  setAppointment: PropTypes.func.isRequired,
};
ConsultancyContainer.defaultProps = {
  authToken: '',
};

const mapStateToPros = state => ({
  consultancies: state.consultanciesReducer.consultancies,
  authToken: state.authenticationReducer.authToken,
});

export default connect(mapStateToPros, { setAppointment })(ConsultancyContainer);
