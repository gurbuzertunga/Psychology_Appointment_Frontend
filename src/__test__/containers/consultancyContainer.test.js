import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ConsultancyContainer from '../../containers/ConsultancyContainer';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <ConsultancyContainer />
  </Provider>,
);

describe('ConsultancyContainer Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.consultancy');
    expect(page).toMatchSnapshot();
  });
});
