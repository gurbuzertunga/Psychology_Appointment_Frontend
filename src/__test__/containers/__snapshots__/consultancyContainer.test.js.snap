// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsultancyContainer Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
